### ISSUE TITLE
It broke!

### ISSUE THREAD

**Issue by johnboiles** [ibigio/shell-ai#25]
I set up a new computer today, and set my `OPENAI_API_KEY` in my `~/.zprofile` and q does not work:

```
> q hello

  Error: Failed to connect to OpenAI.

  API request failed: 404 Not Found
```

I went back to my old computer (which was working great with `shell-ai`) and did a `brew upgrade shell-ai`, and now it does the same thing.

**Comment by ibigio** [comment]
Hm... weird. Just tried reinstalling but not able to replicate. What does your config look like? Maybe try `q config reset`?

**Comment by johnboiles** [comment]
Nope still nothing!
<img width="503" alt="image" src="https://github.com/ibigio/shell-ai/assets/218876/f489b762-8909-4c33-a0b4-55df9a544b44">

Is there any more verbose logging I could enable / look at?

**Comment by ibigio** [comment]
Unfortunately not. Guess you could clone and add them yourself, but that might be a hassle.

Curious to understand which version broke this for you. e.g. if you clone and checkout an older version (`git checkout tags/v1.0.0`), then run `go run main.go hello` and see when it starts working.

And if all else fails I will legit hop on a call to fix this lol.

**Comment by johnboiles** [comment]
I'll try the tags tomorrow but I'm also totally down for a call!

I tried a different shell ai project in the meantime but it was verbose and also slow. I love the minimalism of your tool

**Comment by johnboiles** [comment]
Looks like `v0.3.8` works, but anything newer does not!
```
➜  shell-ai git:(v1.1.0) go run main.go hello

  Error: Failed to connect to OpenAI.

  API request failed: 404 Not Found

➜  shell-ai git:(v1.1.0) git checkout v1.0.0
Previous HEAD position was db552c2 Merge branch 'config'
HEAD is now at 74c64f6 Merge pull request #23 from ibigio/config-release
➜  shell-ai git:(v1.0.0) go run main.go hello

  Error: Failed to connect to OpenAI.

  API request failed: 404 Not Found

➜  shell-ai git:(v1.0.0) git checkout v0.3.8
Previous HEAD position was 74c64f6 Merge pull request #23 from ibigio/config-release
HEAD is now at 4aa328f Updated goreleaser.
➜  shell-ai git:(v0.3.8) go run main.go hello

    echo "hello"
```

**Comment by ibigio** [comment]
Ok I suspected as much – that's when I made a big refactor to support the config file. (Thanks for trying that out!) Are you on windows by any chance? Not sure how this would affect it but wanna make sure.

(Btw in the meantime you can use `go build` in v0.3.8 to make an executable of `q` and then add it to your bin, or edit your path to point to it. So you can keep using it while we fix this.)

**Comment by johnboiles** [comment]
Nope, i'm on macOS. Oooh but I should def put it on my Windows machines because I'm terrible at Powershell/Command Prompt. I wonder what it takes to get a package added to `winget` (similar to `brew` but for Windows, and also officially supported by Microsoft).

Thanks for working on a fix!

**Comment by ibigio** [comment]
...did this miraculously somehow fix itself? 👀 (please?)

**Comment by jerkstorecaller** [comment]
> ...did this miraculously somehow fix itself? 👀 (please?)

I have no such issue in 1.1.0. I tried it Debian 10, Ubuntu 20.04 and 22.04. Everything works fine.

However other people in the community where your tool was linked are complaining about getting the error "sk-XXXXX (the API key) environment variable not set" even though it's clearly set. 
It may be related to locale, idk. I use a default US/english locale.

**Comment by ibigio** [comment]
@jerkstorecaller which community was this tool linked from? Would love to chat with people to get this resolved.

**Comment by jakubklos77** [comment]
This happened to me too but the issue was that I did not put the key in the quotes. Once I fixed this it started working...

**Comment by ibigio** [comment]
Anyone still having issues with this? If so, feel free to re-open!
